### PR TITLE
Cleanup in logic.mods and logic.msbib

### PR DESCRIPTION
**Pull request by oscargus**
Additional change: added method `hasField` to `BibEntry`.

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java [hunk 1]
```diff
@@ -620,28 +621,28 @@ private void getAuthors(Element authorsElem, String bcol) {
...
         }
         return result;
     }
 
     /* construct a MSBib date object */
     protected String getDate(BibEntry bibtex) {
-        String result = "";
-        if (bibtex.getField("year") != null) {
-            result += bibtex.getField("year");
+        StringBuffer result = new StringBuffer();
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java]: I think StringBuilder is enough here (no mulit-threading required)

**Comment by tobiasdiez** [comment]
:+1: 
One could also consider to return an optional from getField but this probably requires more changes.

**Comment by oscargus** [comment]
I was also thinking about that. Would it make sense to add two more methods:
``` java
Optional<String> getOptionalField(String field)
String getDefaultField(String field, String defaultReturnValue)
```

These would make the code look pretty much nicer in quite a few places.

File: src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/msbib/MSBibEntry.java]: I confuse those... I'll have a look!

**Comment by oscargus** [comment]
Ended up adding `getFieldOptional` as the default value can be easily obtained through `orElse`...

**Comment by koppor** [comment]
Why keeping both `getField` and `getFieldOptional`? Why not really replacing the return type of `getField`? That would make `BibEntry.java` consistent within itself. (And I'd remove the `hasField` method, too).

**Comment by oscargus** [comment]
Because getField is used in 600 places... Also hasField should be faster than getField and in enough places one is actually just looking for that information.

**Comment by koppor** [comment]
I'll discuss this at JabCon. :runner:

**Comment by oscargus** [comment]
Do that. There may be refinements to the PR in the meantime. You can only
convert that many getField requests per day...

**Comment by simonharrer** [comment]
Long term: getField should return optional and getFieldOptional should be removed  as this causes confusion with the getOptionalFields
Short term: merge this PR as is

**Pull request merged**